File: 백준/Silver/1730. 판화/판화.py
# https://www.acmicpc.net/problem/1730

n = int(input())
cmds = input()
# 이동방향을 기록함
move_dir_board = [[[] for _ in range(n)] for _ in range(n)]

def in_range(r, c):
    return 0 <= r < n and 0 <= c < n


dir = {'D': (1, 0,),
    'U' : (-1, 0),
    'R' : (0, 1),
    'L' : (0, -1)}


r1, c1 = 0, 0
for cmd in cmds:
    r2, c2 = r1 + dir[cmd][0], c1 + dir[cmd][1]
    # in_range하지 못한 좌표는 명령 무시해야함.
    if in_range(r2, c2):
        # 왜 r1, c1도 방향을 기록해?
        # (0,0)에서 D면 (1,0)이 D방향을 보고 있는 상태임.
        # 즉, 움직임이라는 것은 2칸씩을 쓴다고 볼 수 있음.
        # 그래서 한 움직임 당 2칸씩 move_dir에 기록하고 왼오/상하 서로 다른 소속이면 + 처리
        move_dir_board[r1][c1].append(cmd)
        move_dir_board[r2][c2].append(cmd)
        r1, c1 = r2, c2
        
for r in range(n):
    for c in range(n):
        row = move_dir_board[r][c]
        if ('R' in row or 'L' in row) and ('U' in row or 'D' in row):
            print('+', end='')
        elif ('R' in row or 'L' in row) and not ('U' in row or 'D' in row):
            print('-',end='')
        elif not ('R' in row or 'L' in row) and ('U' in row or 'D' in row):
            print('|',end='')
        else:
            print('.', end='')
    print()

File: 백준/Silver/1730. 판화/판화.java
import java.util.*;

public class Main {

    private static int n;

    private static char[][] upDownBoard;
    private static char[][] leftRightBoard;
    private static char[][] finalBoard;

    public static void main(String[] args) {
        Map<Character, int[]> dir = new HashMap<>();
        dir.put('D', new int[]{1, 0});
        dir.put('U', new int[]{-1, 0});
        dir.put('R', new int[]{0, 1});
        dir.put('L', new int[]{0, -1});

        Scanner sc = new Scanner(System.in);
        
        // Read the size of the board
        n = sc.nextInt();
        sc.nextLine(); // Consume newline character after reading integer
        
        // Read the commands as a string
        String cmds = sc.nextLine();
        
        upDownBoard = new char[n][n];
        leftRightBoard = new char[n][n];

        int cr = 0;
        int cc = 0;

        for (int i = 0; i < cmds.length(); i++) {
            char cmd = cmds.charAt(i);
            int[] directionValues = dir.get(cmd);

            int nr = cr + directionValues[0];
            int nc = cc + directionValues[1];

            if (inRange(nr, nc)) {
                if (cmd == 'D' || cmd == 'U') {
                    upDownBoard[nr][nc] = cmd;
                    upDownBoard[cr][cc] = cmd;
                } else if (cmd == 'R' || cmd == 'L') {
                    leftRightBoard[nr][nc] = cmd;
                    leftRightBoard[cr][cc] = cmd;
                }
                cr = nr;
                cc = nc;
            }
        }

        finalBoard = new char[n][n];
        for (int r = 0; r < n; r++) {
            for (int c = 0; c < n; c++) {
                if (upDownBoard[r][c] == 0 && leftRightBoard[r][c] == 0) {
                    finalBoard[r][c] = '.';
                } else if (upDownBoard[r][c] != 0 && leftRightBoard[r][c] != 0) {
                    finalBoard[r][c] = '+';
                } else if (upDownBoard[r][c] != 0 && leftRightBoard[r][c] == 0) {
                    finalBoard[r][c] = '|';
                } else if (upDownBoard[r][c] == 0 && leftRightBoard[r][c] != 0) {
                    finalBoard[r][c] = '-';
                }
            }
        }

        // Print the final board
        for (int r = 0; r < n; r++) {
            for (int c = 0; c < n; c++) {
                System.out.print(finalBoard[r][c]);
            }
            System.out.println();
        }

        // Close the scanner
        sc.close();
    }

    private static boolean inRange(int r, int c) {
        return 0 <= r && r < n && 0 <= c && c < n;
    }
}